apps/container/src/views/HistoryView.vue

```diff
@@ -246,19 +266,48 @@
 }
 </style>
 <script setup lang="ts">
+import { useQuery } from '@tanstack/vue-query';
+import { conceptsColor } from 'consts';
+import { Enrollment, Enrollments, Users } from 'services';
+import { Concept } from 'types';
+import { computed, ref } from 'vue';
+import ReviewDialog from '@/components/ReviewDialog.vue';
+import CenteredLoading from '@/components/CenteredLoading.vue';
 import PaperCard from '@/components/PaperCard.vue';
 import TableComponent from '@/components/TableComponent.vue';
-import {
-  Users,
-  Enrollments,
-  Enrollment as EnrollmentType,
-  Concept,
-} from 'services';
-import { computed } from 'vue';
-import CenteredLoading from '@/components/CenteredLoading.vue';
 import FeedbackAlert from '@/components/FeedbackAlert.vue';
-import { ref } from 'vue';
-import { useQuery } from '@tanstack/vue-query';
+import { checkEAD } from 'utils';
+
+const showDialog = ref(false);
+const selectedEnrollment = ref<Enrollment | undefined>();
+
+const tags = ref<string[]>([]);
+const handleOpenDialog = (
+  enrollment: Enrollment,
+  type: 'teoria' | 'pratica',
+) => {
+  const processedEnrollment: Enrollment = {
+    ...enrollment,
+    teoria: type !== 'pratica' ? enrollment.teoria : null,
+    pratica: type === 'pratica' ? enrollment.pratica : null,
+  };
+  selectedEnrollment.value = processedEnrollment;
+
+  const isEAD = checkEAD(enrollment.year || 0, enrollment.quad || 0);
+
+  const processedTags = [
+    enrollment.pratica?._id === enrollment.teoria?._id
+      ? 'teoria e prática'
+      : type === 'pratica'
+      ? 'prática'
+      : 'teoria',
+    `Q${processedEnrollment.quad} ${processedEnrollment.year}`,
+  ];
+  isEAD && processedTags.push('EAD');
+  tags.value = processedTags;
```

Comment:
acha que assim facilita a leitura?
```suggestion
tags.value = [
    enrollment.pratica?._id === enrollment.teoria?._id && 'teoria e prática',
    type === 'pratica' && 'prática',
    type === 'teoria' && 'teoria',
    `Q${processedEnrollment.quad} ${processedEnrollment.year}`,
    isEAD && 'EAD'
  ].filter(Boolean) as string[];
```

Reply:
Facilita mas n dá o mesmo resultado

Reply:
Peguei ideias desse que vc fez e fiz um novo